pynest/nest/hl_api.py

```diff
@@ -23,35 +23,31 @@
 High-level API of PyNEST Module
 """
 
-import os
-import sys
-
 # We search through the subdirectory "lib" of the "nest" module
 # directory and import the content of all Python files therein into
 # the global namespace. This makes the API functions of PyNEST itself
 # and those of extra modules available to the user.
 
 
-def _pkg_attrs(pkg):
-    # return all package attributes that
-    # from $pkg import *
-    # would put in the module namespace $pkg
-    # see https://docs.python.org/3/reference/simple_stmts.html#import
-    try:
-        # if there's an "__all__", return that
-        return pkg.__all__
-    except AttributeError:
-        # otherwise, return everything at top level that doesn't start with `_`
-        return (attr for attr in pkg.__dict__ if attr[0] != '_')
-
+def _import_libs(mod_file, mod_dict, path,
+                 prefix=None, ignore=frozenset(), level=1):
+    """
+    _import_libs: construct a relative import of all modules
+      mod_file: name of calling module file (__file__)
+      mod_dict: globals() of calling module
+      path: relative path to check for modules
+      prefix: string for import prefix (defaults to path)
+      ignore: set of import names ($prefix.$name) to not import
+      level: relative level (1 is relative to path, 2 is ../path, ...)
+             (default 1)
+      """
 
-def _import_names(pkg, mod_dict):
-    # import * names from pkg to the dictionary of mod_dict
-    for attr in _pkg_attrs(pkg):
-        mod_dict[attr] = getattr(pkg, attr)
+    import ast
+    import os
```

Comment:
According to PEP 8, imports are always put at the top of the file. (https://www.python.org/dev/peps/pep-0008/#imports)

Reply:
Should I pollute the hl_api namespace with unneeded imports? I think the goal of the pep8 requirement is to put global imports at the top of the file rather than in the middle or at the bottom, for ease of reading (but who knows -- a lot of pep8 is fetishes by weird folks).

Another option is to extract all of this and put it in a submodule to hl_api. Probably cleaner that way.

Reply:
I agree with you avoiding hl_api namespaces pollution. It is not necessary to put import on global level. Thus, I approved the PR.